packages/react-router5/index.d.ts

```diff
@@ -37,22 +37,23 @@ declare module 'react-router5' {
 
     export const Link: ComponentClass<LinkProps>
 
-    export type InjectedRoute = Partial<{
+    export type InjectedRoute = {
+        router: Router
         route: State
         previousRoute: State
-    }>
+    }
 
-    export function withRoute<TProps extends InjectedRoute>(
+    export function withRoute<TProps extends Partial<InjectedRoute>>(
         BaseComponent: ComponentType<TProps>
     ): ComponentClass<Omit<TProps, keyof InjectedRoute>>
 
-    export type InjectedRouterNode = Partial<{
+    export type InjectedRouterNode = {
         router: Router
         previousRoute: State
```

Comment:
Ditto.

Reply:
Sorry, I don't get it :-) What's wrong here?

Reply:
Ditto just means to repeat the last thing I mentioned, which was whether this is guaranteed or not.